drivers/timer/nrf_rtc_timer.c

```diff
@@ -386,11 +386,7 @@ uint32_t sys_clock_elapsed(void)
 		return 0;
 	}
 
-	k_spinlock_key_t key = k_spin_lock(&lock);
-	uint32_t ret = counter_sub(counter(), last_count) / CYC_PER_TICK;
-
-	k_spin_unlock(&lock, key);
```

Comment:
This looks wrong.  That comment isn't saying you don't need synchronization, it's saying that this function isn't reentrant: you don't need to worry about other users of the timeout system asking for the current time simultaneously.

In fact that last_count variable is mutated from an ISR, so if you aren't locking interrupts this can race, e.g.:

* Call `counter()`, which returns a time 1 cycle before the expiration value "`T`"
* The counter advances and the interrupt preempts, then advances last_count to be equal to `T`
* The ISR returns and this subtracts last_count (now "`T`") from counter()'s result from earlier ("`T - 1`"), giving `0xffffff`
* The world blows up due to the overflowed value

Maybe I should remove that sentence in the docs.